bluepyopt/ephys/create_acc.py

```diff
@@ -0,0 +1,832 @@
+                                          pprocess_mechs)
+
+    ret = {filenames[name]:
+           template.render(template_name=template_name,
+                           banner=banner,
+                           morphology=morphology,
+                           replace_axon=replace_axon_path,
+                           modified_morphology=modified_morphology_path,
+                           filenames=filenames,
+                           label_dict=label_dict,
+                           global_mechs=global_mechs,
+                           global_scaled_mechs=global_scaled_mechs,
+                           local_mechs=local_mechs,
+                           local_scaled_mechs=local_scaled_mechs,
+                           pprocess_mechs=pprocess_mechs,
+                           **custom_jinja_params)
+           for name, template in templates.items()}
+
+    if replace_axon is not None:
+        ret[replace_axon_path] = replace_axon_acc
+        if modified_morphology_path is not None:
+            ret[modified_morphology_path] = modified_morphology_acc
+
+    return ret
+
+
+def write_acc(output_dir, cell, parameters,
+              template_filename='acc/*_template.jinja2',
+              ext_catalogues=None,
+              create_mod_morph=False,
+              sim=None):
+    '''Output mixed JSON/ACC format for Arbor cable cell to files
+
+    Args:
+        output_dir (str): Output directory. If not exists, will be created
+        cell (): Cell model to output
+        parameters (): Values for mechanism parameters, etc.
+        template_filename (str): file path of the cell.json , decor.acc and
+        label_dict.acc jinja2 templates (with wildcards expanded by glob)
+        ext_catalogues (): Name to path mapping of non-Arbor built-in
+        NMODL mechanism catalogues compiled with modcc
+        create_mod_morph (str): Output ACC with axon replacement
+        sim (): Neuron simulator instance (only used used with axon
+        replacement if morphology has not yet been instantiated)
+    '''
+    output = cell.create_acc(parameters, template=template_filename,
+                             ext_catalogues=ext_catalogues,
+                             create_mod_morph=create_mod_morph,
+                             sim=sim)
+
+    cell_json = [comp_rendered
+                 for comp, comp_rendered in output.items()
+                 if pathlib.Path(comp).suffix == '.json']
+    if len(cell_json) != 1:
+        raise CreateAccException(
+            'JSON file from create_acc is non-unique: %s' % cell_json)
+
+    cell_json = json.loads(cell_json[0])
+
+    output_dir = pathlib.Path(output_dir)
+    if not output_dir.exists():
+        output_dir.mkdir()
+    for comp, comp_rendered in output.items():
+        comp_filename = output_dir.joinpath(comp)
+        if comp_filename.exists():
+            raise CreateAccException("%s already exists!" % comp_filename)
```

Comment:
Naming exceptions after modules is the current style in the rest of BluePyOpt. If you propose a new exception organization, I'd suggest to do that in a new PR for all the library at once instead of here only on the Arbor exporter code. Otherwise, please go ahead and change it.

Reply:
No, I prefer newly added exception types to be as informative as possible.

Here when the file does not exist I would like to see a FileExistsError. I only would like to see CreateAcc exception when the error really is due to Arbor internals.

We decided we (maintainers) only comment from now on, no? You change please :)

Reply:
Sure, the exception policy is smth you decide at the library level. Since it's one coherent task and you'll continue on this outside of this PR, I'd be glad if you could make your changes directly on the code.